Core/Src/sensors/steering.c

```diff
@@ -29,40 +41,41 @@ void steering_read(void* argument) {
 
         double zero_aux = ZERO_VOLANTE;
 
-        // Se o minimo do volante for menor que 0, o sensor voltara no valor maximo do ADC
-        // se isso acontecer, o valor do ADC voltara para 4095
-        // entao subtrai 4095 do valor lido, dando um valor negativo que pode ser aplicado
-        // na formula o mesmo vale pro zero do volante
-        if (VOLANTE_MIN > VOLANTE_MAX) {
+        // Como o sensor nao possui batente, se o minimo do volante for menor que o 0 do ADC,
+        // o sensor voltara no valor maximo do ADC (4095)
+        // entao é subtraido 4095 do valor lido, dando um valor negativo que pode ser aplicado
+        // na formula
+        if (ZERO_VOLANTE > VOLANTE_MAX) {
             zero_aux -= 4095;
             if (volante_cru > VOLANTE_MAX) {
                 volante_cru -= 4095;
             }
         }
 
-        if (volante_cru < zero_aux) {
-            set_global_var_value(STEERING_WHEEL, 0);
-        } else {
-            set_global_var_value(STEERING_WHEEL,
-                                 (uint16_t)volante_cru * GANHO_VOLANTE - ZERO_VOLANTE);
-        }
 
-        STEERING_WHEEL_t steering_wheel = get_global_var_value(STEERING_WHEEL);
-
-        log_data(ID_STEERING_WHEEL, steering_wheel);
 
+        // A funcao converte o valor do ADC para o valor em graus do estercamento do volante
+        // Tambem informa a direcao do volante
         // SPAN_ALINHAMENTO eh apenas um span pra ainda considerar o volante no centro
         // ate uma certa quantidade
-        if (steering_wheel > VOLANTE_ALINHADO + SPAN_ALINHAMENTO) {
+        if (volante_cru > VOLANTE_ALINHADO + SPAN_ALINHAMENTO) {
             set_global_var_value(INTERNAL_WHEEL, ESQUERDA);
-        } else if (steering_wheel < VOLANTE_ALINHADO - SPAN_ALINHAMENTO) {
+            set_global_var_value(STEERING_WHEEL,
+                                       (uint16_t)(calculo_steering(volante_cru, ZERO_VOLANTE, VOLANTE_MAX, ANG_MAX_VOLANTE) - ANGULO_VOLANTE_ALINHADO));
+        } else if (volante_cru < VOLANTE_ALINHADO - SPAN_ALINHAMENTO) {
             set_global_var_value(INTERNAL_WHEEL, DIREITA);
+            set_global_var_value(STEERING_WHEEL,
+            							(uint16_t)ANGULO_VOLANTE_ALINHADO - calculo_steering(volante_cru, ZERO_VOLANTE, VOLANTE_MAX, ANG_MAX_VOLANTE));
         } else {
             set_global_var_value(INTERNAL_WHEEL, CENTRO);
+            set_global_var_value(STEERING_WHEEL, 0);
         }
 
+        STEERING_WHEEL_t steering_wheel = get_global_var_value(STEERING_WHEEL);
+
+        log_data(ID_STEERING_WHEEL, steering_wheel);
+
         osDelay(100);
```

Comment:
coloca esse 100 em um define que chama STEERING_CALC_DELAY.
pode reduzir pra 50 tbm, não vai fazer mal

Reply:
Esse define seria na própria task?

Reply:
No início do arquivo steering.h